var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -201,8 +201,8 @@ def setup_run_environment(self, env):
         env.set("DD4HEP", self.prefix.examples)
         env.set("DD4hep_DIR", self.prefix)
         env.set("DD4hep_ROOT", self.prefix)
-        if len(self.libs.directories) > 0:
-            env.prepend_path("LD_LIBRARY_PATH", self.libs.directories[0])
+
+        # Note: ROOT dependency automatically sets up ROOT environment vars
```

Comment:
Same comment as below, DD4hep searches in `LD_LIBRARY_PATH`

Reply:
Does dd4hep search for its own libraries in this path?

Reply:
DD4hep just uses plain dlopen, https://github.com/AIDASoft/DD4hep/blob/abb85cbacd3909ea54733f2defd8053eaee63d12/DDCore/src/Plugins.cpp#L117, so it implicitly uses `LD_LIBRARY_PATH`.

Reply:
Yes, quite a large fraction of the DD4hep (core) functionality is also implemented as plugins:
```console
ls $DD4hep_DIR/lib/*.components
$DD4hep_DIR/lib/libDDAlignPlugins.components
$DD4hep_DIR/lib/libDDCADPlugins.components
$DD4hep_DIR/lib/libDDCondPlugins.components
$DD4hep_DIR/lib/libDDCoreGraphics.components
$DD4hep_DIR/lib/libDDCorePlugins.components
$DD4hep_DIR/lib/libDDDetectors.components
$DD4hep_DIR/lib/libDDDigiPlugins.components
$DD4hep_DIR/lib/libDDDigi_IO.components
$DD4hep_DIR/lib/libDDG4EDM4HEP.components
$DD4hep_DIR/lib/libDDG4HepMC3.components
$DD4hep_DIR/lib/libDDG4LCIO.components
$DD4hep_DIR/lib/libDDG4Plugins.components
$DD4hep_DIR/lib/libDDG4Python.components
$DD4hep_DIR/lib/libDDGDMLPlugins.components
$DD4hep_DIR/lib/libDDPythonPlugins.components
$DD4hep_DIR/lib/libDDRecPlugins.components
```